Sources/Leaf/Request+Leaf.swift

```diff
@@ -2,16 +2,16 @@ import Vapor
 
 extension Request {
     var leaf: LeafRenderer {
-        .init(
+        self.application.leaf.userInfo["request"] = self
```

Comment:
This should create a copy of the application's user info before injecting the request into it. Otherwise we are holding onto the Request object from the Application.